api/swagger.yaml

```diff
@@ -77,43 +91,341 @@ tags:
+
+      ### startWaitingEvent, stopWaitingEvent
+
+      Marks the moment a `Vehicle` starts or stops waiting.
+
+      _Applicable entities for `startWaitingEvent` and `stopWaitingEvent`:_
+
+      | Location | Vehicle | Shipment | Trip  | Route | Actor | Constraint | Sensor | Event |
+      | :------: | :-----: | :------: | :---: | :---: | :---: | :--------: | :----: | :---: |
+      |          | ●       |          |       |       |       |            |        |       |
+
+      ### startLoadingAndUnloadingEvent, stopLoadingAndUnloadingEvent
+
+      Marks the start or end of a time period meant for loading and unloading.
+
+      _Applicable entities for `startLoadingAndUnloadingEvent` and `stopLoadingAndUnloadingEvent`:_
+
+      | Location | Vehicle | Shipment | Trip  | Route | Actor | Constraint | Sensor | Event |
+      | :------: | :-----: | :------: | :---: | :---: | :---: | :--------: | :----: | :---: |
+      |          | ●       |          |       |       |       |            |        |       |
+
+      ## Time window events
+
+      Time window events are used to model a time window in a `planned` or `projected` lifecycle phase. The start and end of a time window are
+      modelled as separate events. The following time window events are available:
+
+      * `pickupTimeWindowStartEvent`: Marks the start of a time window for the pickup of a `Shipment`.
+      * `pickupTimeWindowEndEvent`: Marks the end of a time window for the pickup of a `Shipment`.
+      * `deliveryTimeWindowStartEvent`: Marks the start of a time window for the delivery of a `Shipment`.
+      * `deliveryTimeWindowEndEvent`: Marks the end of a time window for the delivery of a `Shipment`.
+
+      _Applicable lifecycle phases for time window events:_
+
+      | Planned | Projected | Actual | Realized |
+      | :-----: | :-------: | :----: | :------: |
+      |  ●      | ●         |        |          |
+
+      _Applicable entities for `pickupTimeWindowStartEvent`, `pickupTimeWindowEndEvent`, `deliveryTimeWindowStartEvent` and
+      `deliveryTimeWindowEndEvent`:_
+
+      | Location | Vehicle | Shipment | Trip  | Route | Actor | Constraint | Sensor | Event |
+      | :------: | :-----: | :------: | :---: | :---: | :---: | :--------: | :----: | :---: |
+      |          |         | ●        |       |       |       |            |        |       |
+
+      ## Associative events
+
+      Associative events are used to create or remove associations between entities. These events can take place in all lifecycle phases.
+      Associative events typically involve two events.
+
+      _Applicable lifecycle phases for associative events:_
+
+      | Planned | Projected | Actual | Realized |
+      | :-----: | :-------: | :----: | :------: |
+      |  ●      | ●         | ●      | ●        |
+
+      ### coupleVehiclesEvent, decoupleVehiclesEvent
+
+      Marks the moment two `Vehicle`s are coupled or decoupled.
```

Comment:
Why would you actually want to couple vehicles, for platooning, or is there another use case?